syntax/crisp_lexer.mll

```diff
@@ -11,6 +11,40 @@
 open Lexing
 open Crisp_parser
 
+exception Error
+
+let backtrack str lexbuf =
+  lexbuf.lex_curr_pos <- lexbuf.lex_curr_pos - (String.length str)
+
+let count_nl str =
+  if str <> (String.make (String.length str) '\n') then
+    failwith "Counting new lines in a string containing not only new lines"
+  else
+    String.length str (*NOTE assumes we only use unix-style newlines*)
+
+let next_line ?(nl_count=1) ?nl_str ?(nl_at_end=false) lexbuf =
```

Comment:
Since this function has become more complex it's useful to add a one or two-sentence comment on the role it plays in the larger machinery, so when reading the code the programmer will have a better idea of what to expect.